src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -1943,12 +1943,30 @@ public TypeKind getKind() {
          */
         public Type lower;
 
+        /** if this type variable is universal then it could also have a link to a pure reference
+         *  type variable, it is important to know that a universal type variable and its
+         *  corresponding referenceTypeVar share the same tsym. So if it is needed to double check if
+         *  a type variable is universal or not, we need to check its type not the type of its tsym
+         */
+        public TypeVar projection = null;
+
+        public boolean isReferenceProjection;
+
+        // redundant for now but helpful for debug reasons
+        public boolean isUniversal;
+
         public TypeVar(Name name, Symbol owner, Type lower) {
+            this(name, owner, lower, false, false);
+        }
+
+        public TypeVar(Name name, Symbol owner, Type lower, boolean isUniversal, boolean isReferenceProjection) {
```

Comment:
This constructor probably should not take the `isReferenceProjection` as this one creates a new tsym and you want the reference projection to share the same symbol - so you probably are always passing `false` here.